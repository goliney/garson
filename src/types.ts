export type keyType = string;

export type onSubmitCallback = (result: any) => void;

export type promptCallback = (onSubmit: onSubmitCallback) => JSX.Element;

export type promptKeyPairs = PromptKeyPair[];

// action should return either nothing or another garson config
export type actionCallback = (results: Results) => GarsonConfig | void;

export interface PromptKeyPair {
  key: keyType;
  prompt: promptCallback;
}

export interface Results {
  [key: string]: any;
}

export interface GarsonHandlerParams {
  results?: Results;
  prompts?: promptKeyPairs;
}

export interface GarsonConfig {
  results?: Results;
  prompts: promptKeyPairs;
  action: actionCallback;
}
