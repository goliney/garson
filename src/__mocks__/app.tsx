import React from 'react';
import { render } from 'ink-testing-library';
import { Text } from 'ink';

const initialView = <Text>Initializing tests...</Text>;

export const app = render(initialView);

app.unmount = jest.fn().mockImplementation(() => app.rerender(initialView));
