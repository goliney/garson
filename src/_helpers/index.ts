// Components
export { Question } from './components/question';
export { Line } from './components/line';

// Hooks
export { useKeyHandler } from './use-key-handler';
export { useEnterKeyHandler } from './use-enter-key-handler';
