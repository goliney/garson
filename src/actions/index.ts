import { spawn } from './spawn';

export const actions = {
  spawn,
};
