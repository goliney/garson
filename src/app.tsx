import { Box, render } from 'ink';
import React from 'react';

export const app = render(<Box>Initializing...</Box>, {
  debug: false,
});
