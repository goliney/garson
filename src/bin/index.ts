import { runner } from './runner';

const cwd = process.cwd();
let garsonConfig;

try {
  // eslint-disable-next-line import/no-dynamic-require, global-require
  garsonConfig = require(`${cwd}/garson.config.js`);
} catch (e) {
  console.error('Failed to load config', e);
}

if (garsonConfig) {
  runner(garsonConfig);
}
