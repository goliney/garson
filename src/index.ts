import { KeyType, PromptCallback, Results, ActionCallback, GarsonHandlerParams } from './types';

export function garson(results?: Results) {
  return garsonHandler({ results });
}

function garsonHandler({ results = {}, prompts = [] }: GarsonHandlerParams) {
  return {
    prompt: (key: KeyType, prompt: PromptCallback) =>
      garsonHandler({ results, prompts: [...prompts, { key, prompt }] }),
    action: (action: ActionCallback) => ({
      results,
      prompts,
      action,
    }),
  };
}
