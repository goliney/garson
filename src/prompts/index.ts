export { input } from './input';
export { choices } from './choices';
export { fuzzyPath } from './fuzzy-path';
