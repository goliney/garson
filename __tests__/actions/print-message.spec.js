import { actions } from '../../src';
import { app } from '../../src/app';

jest.mock('../../src/app');

describe('Print Message', () => {
  beforeEach(() => {
    process.stdout.columns = 80; // needed for line output to be consistent
  });
  afterEach(() => {
    app.unmount();
  });

  test('With box title', async () => {
    actions.printMessage({ message: `Hello`, boxTitle: 'Simon says' });
    expect(app.lastFrame()).toMatchSnapshot();
  });

  test('Without box title', async () => {
    actions.printMessage({ message: `Simon says hi` });
    expect(app.lastFrame()).toMatchSnapshot();
  });
});
