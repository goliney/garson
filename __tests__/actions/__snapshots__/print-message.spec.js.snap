// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print Message With box title 1`] = `
"[38;2;0;128;0mSimon says=====================================================================[39m

[3mHello[23m

[38;2;0;128;0m===============================================================================[39m
"
`;

exports[`Print Message Without box title 1`] = `
"[38;2;0;128;0m===============================================================================[39m

[3mSimon says hi[23m

[38;2;0;128;0m===============================================================================[39m
"
`;
